client/src/components/GroupsTable.js

```diff
@@ -0,0 +1,283 @@
+import { useEffect, useState, useCallback } from 'react';
+import PropTypes from 'prop-types';
+import {
+  Box,
+  Table,
+  TableBody,
+  TableCell,
+  TableContainer,
+  TableHead,
+  TablePagination,
+  TableRow,
+  TableSortLabel,
+  Typography,
+  Toolbar,
+  Button
+} from '@mui/material';
+import { Link } from 'react-router-dom';
+import Paper from '@mui/material/Paper';
+import { visuallyHidden } from '@mui/utils';
+
+function descendingComparator(a, b, orderBy) {
+  if (b[orderBy] < a[orderBy]) {
+    return -1;
+  }
+  if (b[orderBy] > a[orderBy]) {
+    return 1;
+  }
+  return 0;
+}
+
+function getComparator(order, orderBy) {
+  return order === 'desc'
+    ? (a, b) => descendingComparator(a, b, orderBy)
+    : (a, b) => -descendingComparator(a, b, orderBy);
+}
+
+function stableSort(array, comparator) {
+  const stabilizedThis = array?.map((el, index) => [el, index]);
+  if (stabilizedThis?.length) {
+    stabilizedThis.sort((a, b) => {
+      const order = comparator(a[0], b[0]);
+      if (order !== 0) {
+        return order;
+      }
+      return b[1] - a[1];
+    });
+    return stabilizedThis.map((el) => el[0]);
+  }
+}
```

Comment:
This code looks familiar!

Think about extracting these comparator and sort functions as utilities that can be reused.